test/test_retry.py

```diff
@@ -271,3 +274,47 @@ def test_retry_set_remove_headers_on_redirect(self):
         retry = Retry(remove_headers_on_redirect=["X-API-Secret"])
 
         assert list(retry.remove_headers_on_redirect) == ["x-api-secret"]
+
+    @pytest.mark.parametrize("value", ["-1", "+1", "1.0", six.u("\xb2")])  # \xb2 = ^2
+    def test_parse_retry_after_invalid(self, value):
+        retry = Retry()
+        with pytest.raises(InvalidHeader):
+            retry.parse_retry_after(value)
+
+    @pytest.mark.parametrize(
```

Comment:
Can we add a test case for an http-date formatted Retry-After value? May require mocking time.time() to get a consistent result.

Reply:
Added four cases here (whether the time is in the future or not X whether we're respecting the header or not).